### ISSUE TITLE
Wrong UI in EKS cluster edit

### ISSUE THREAD

**Issue by giuseppeCozzolino** [rancher/rancher#32004]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create EKS cluster with custom Subnets/VPC and custom Security Groups
- Edit the cluster
- Subnets are not editable and the UI shows the wrong subnets selection (Standard: VPC and Subnets are automatically created)

**Result:**
- Show in the UI the previously selected subnets

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.6, 2.5.7
- Installation option (single install/HA): single install and HA
<img width="1491" alt="Schermata 2021-04-09 alle 12 19 02" src="https://user-images.githubusercontent.com/77336342/114166697-5e071580-992e-11eb-8a77-188527003a52.png">

**Comment by aaronRancher** [comment]
Reproduced this issue on `v2.5.7` and `v2.5.5`.

Steps:
Create EKS cluster with custom Subnets/VPC and custom Security Groups
Edit the cluster

Result:
Custom subnet is not shown: `Standard: VPC and Subnet are automatically created`

During cluster provisioning and before the EKS operator sync, the correct subnets are still shown on edit:
![image](https://user-images.githubusercontent.com/57409052/114221838-a91b2b80-9922-11eb-9cf0-127b877d3130.png)

**Comment by westlywright** [comment]
@aaronRancher After the EKS operator sync does the data in `upstreamSpec` on the cluster object match what is in the config? If something is happening to the data after the sync occurs this wouldn't be a UI issue.

**Comment by aaronRancher** [comment]
@westlywright The `upstreamSpec` matches the config after EKS operator sync. Subnets are still seen in `upstreamSpec`.
```
"upstreamSpec": {
"amazonCredentialSecret": "cattle-global-data:cc-k75sv",
"displayName": "aaron-test4",
"imported": false,
"kmsKey": "",
"kubernetesVersion": "1.18",
"loggingTypes": [ ],
"nodeGroups": [
...
},
"maxSize": 2,
"minSize": 2,
"nodegroupName": "aaron-ng",
"requestSpotInstances": false,
"resourceTags": null,
"spotInstanceTypes": null,
"subnets": [ 2 items
"subnet-1ed44d64",
"subnet-6127e62d"
],
...
"region": "us-east-2",
"secretsEncryption": false,
"securityGroups": [
"sg-0aa35aee3658b0e2a"
],
"serviceRole": "",
"subnets": [ 2 items
"subnet-1ed44d64",
"subnet-6127e62d"
],
"tags": { },
"type": "/v3/schemas/eksClusterConfigSpec"
},
```

**Comment by aaronRancher** [comment]
This bug fix is verified with `v2.5-head` `version 41f173a` and `master-head` `version 1c3629f`

Previously selected subnets are visible on cluster edit.
If the user does not use a preexisting subnet, `Rancher created VPC and Subnet` is seen on edit.
